packages/catalog-platformstate-writer/src/consumerServiceV1.ts

```diff
@@ -1,12 +1,162 @@
 import { match } from "ts-pattern";
-import { EServiceEventEnvelopeV1 } from "pagopa-interop-models";
+import {
+  Descriptor,
+  descriptorState,
+  EServiceDescriptorV1,
+  EServiceEventEnvelopeV1,
+  EServiceId,
+  fromDescriptorV1,
+  makeGSIPKEServiceIdDescriptorId,
+  makePlatformStatesEServiceDescriptorPK,
+  missingKafkaMessageDataError,
+  PlatformStatesCatalogEntry,
+  unsafeBrandId,
+} from "pagopa-interop-models";
 import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import {
+  deleteCatalogEntry,
+  descriptorStateToItemState,
+  readCatalogEntry,
+  updateDescriptorStateInPlatformStatesEntry,
+  updateDescriptorStateInTokenGenerationStatesTable,
+  writeCatalogEntry,
+} from "./utils.js";
 
 export async function handleMessageV1(
   message: EServiceEventEnvelopeV1,
-  _dynamoDBClient: DynamoDBClient
+  dynamoDBClient: DynamoDBClient
 ): Promise<void> {
   await match(message)
+    .with({ type: "EServiceDescriptorUpdated" }, async (msg) => {
+      const eserviceId = unsafeBrandId<EServiceId>(msg.data.eserviceId);
+      const descriptor = parseDescriptor(msg.data.eserviceDescriptor, msg.type);
+
+      const eserviceDescriptorPK = makePlatformStatesEServiceDescriptorPK({
+        eserviceId,
+        descriptorId: descriptor.id,
+      });
+      await match(descriptor.state)
+        .with(descriptorState.published, async () => {
+          const existingCatalogEntry = await readCatalogEntry(
+            eserviceDescriptorPK,
+            dynamoDBClient
+          );
+
+          if (
+            existingCatalogEntry &&
```

Comment:
Here you can apply the same changes I highlighted [in the other PR](https://github.com/pagopa/interop-be-monorepo/pull/942#discussion_r1777134570)

Reply:
Updated